sdk/logs/src/main/java/io/opentelemetry/sdk/logs/ReadWriteLogRecord.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.logs;
+
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.sdk.logs.data.LogData;
+
+/** A log record that can be read from and written to. */
+public interface ReadWriteLogRecord {
+
+  /**
+   * Sets an attribute on the log record. If the log record previously contained a mapping for the
+   * key, the old value is replaced by the specified value.
+   *
+   * <p>Note: the behavior of null values is undefined, and hence strongly discouraged.
+   */
+  <T> ReadWriteLogRecord setAttribute(AttributeKey<T> key, T value);
```

Comment:
@trask for now the only thing you can mutate is log attributes. I plan on adding getters / setters on here in the future, but wondering if this is sufficient for your short term requirements.

Reply:
yes, this is exactly what I need for now 👍